### PR TITLE
e2e: klog output

### PR DESCRIPTION
**Pull request by pohly**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Several improvements to how e2e.test handles klog output. See commit messages for details.

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

### CODE REVIEW

File: test/e2e/framework/log_test.go [hunk 1]
```diff
@@ -31,11 +36,6 @@ import (
 // Here are some intentionally blank lines that can be removed to compensate
 // for future additional import statements.
 //
-//
```
Comment (aojea) [inline test/e2e/framework/log_test.go:34]: heh, almost consumed the buffer :smile:

File: test/e2e/e2e.go [hunk 2]
```diff
@@ -97,8 +97,18 @@ var _ = ginkgo.SynchronizedAfterSuite(func() {
...
+	// in the E2E test suite because it has no feature gates. It used to have a
+	// misleading --feature-gates parameter but that didn't do what users
+	// and developers expected (define which features the cluster supports)
+	// and therefore got removed.
+	//
+	// Because contextual logging is useful and should get tested, it gets
+	// re-enabled here unconditionally.
 	logs.InitLogs()
 	defer logs.FlushLogs()
+	klog.EnableContextualLogging(true)
```
Comment (aojea) [inline test/e2e/e2e.go:110]: are we sure about this? can we postpone this for next week?
we are going to merge the hooks changes and I'm afraid of having too many changes at the same time that will be hard to bisect later

Comment (pohly) [reply test/e2e/e2e.go:110]: Sure, we can wait with this.

**Comment by pohly** [comment]
@aojea : can we merge this now?

File: test/e2e/framework/internal/output/output.go [hunk 3]
```diff
@@ -123,6 +123,9 @@ var stackLocation = regexp.MustCompile(`(?:/|vendor/|test/|GOROOT/).*/([[:^space
 // functionArgs matches "<function name>(...)".
 var functionArgs = regexp.MustCompile(`([[:alpha:]]+)\(.*\)`)
 
+// klogPrefix matches "I0822 16:10:39.343790  989127 "
+var klogPrefix = regexp.MustCompile(`(?m)^[IEF][[:digit:]]{4} [[:digit:]]{2}:[[:digit:]]{2}:[[:digit:]]{2}\.[[:digit:]]{6}[[:space:]]+[[:digit:]]+ `)
```
Comment (aojea) [inline test/e2e/framework/internal/output/output.go:127]: I see  https://github.com/kubernetes/klog/blob/cb9292a1800659470ad0587b2fe18a30ce2ece7e/klog.go#L631-L633

```
/*
header formats a log header as defined by the C++ implementation.
It returns a buffer containing the formatted header and the user's file and line number.
The depth specifies how many stack frames above lives the source line to be identified in the log message.
Log lines have this form:
	Lmmdd hh:mm:ss.uuuuuu threadid file:line] msg...
where the fields are defined as follows:
	L                A single character, representing the log level (eg 'I' for INFO)
	mm               The month (zero padded; ie May is '05')
	dd               The day (zero padded)
	hh:mm:ss.uuuuuu  Time in hours, minutes and fractional seconds
	threadid         The space-padded thread ID as returned by GetTID()
	file             The file name
	line             The line number
	msg              The user-supplied message
*/
```
non warnings ? [IEF]

**Comment by aojea** [comment]
/lgtm
/approve

It seems part of the KEP https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/3077-contextual-logging#logging-in-tests , and merging today gives us the weekend (that is less noise) to notice the impact (slowness, ...)

/hold
/cc @liggit @dims 

for his opinion, feel free to unhold

File: test/e2e/framework/internal/output/output.go (hunk 3 above)
Comment (pohly) [reply test/e2e/framework/internal/output/output.go:127]: I wasn't trying to be complete as this is just for testing purposes. Those only use error and info.

**Comment by aojea** [comment]
/hold cancel

taking it as lazy consensus

**Comment by pohly** [comment]
/retest

Unrelated flake ("Kubernetes e2e suite: [It] [sig-node] Probing container should mark readiness on pods to false and disable liveness probes while pod is in progress of terminating" in pull-kubernetes-kind).

**Pull request merged**